### PR TITLE
Implement isclose

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented isclose.
I want to discuss its specification for scalar values. When input values are scalars, numpy returns `bool` object. But in cupy, we need to transfer the data. I use `cupy.ndarray` in such case in this PR same as `cupy.sum`. How do you think about this problem?

### CODE REVIEW

File: cupy/logic/comparison.py [hunk 1]
```diff
@@ -1,10 +1,57 @@
...
+    xfin = cupy.isfinite(x)
+    yfin = cupy.isfinite(y)
+    if all(xfin) and all(yfin):
+        cond = within_tol(x, y)
+    else:
+        finite = xfin & yfin
+        cond = cupy.zeros_like(finite)
+        # For boolean indexing, we need to broadcast arrays.
+        x, y = cupy.broadcast_arrays(x, y)
+        cond[finite] = within_tol(x[finite], y[finite])
```
Comment (okuta) [inline cupy/logic/comparison.py]: I think this code is a little slow.
How about using Elementwise or ufunc?

Comment (unnonouno) [reply cupy/logic/comparison.py]: OK. As you say i can easily rewrite it with a single element-wise function.

**Comment by unnonouno** [comment]
merge #789 first

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by okuta** [comment]
Please check test result.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by okuta** [comment]
LGTM!

**Pull request merged**